controllers/lxd_initializer_ds.go

```diff
@@ -214,13 +213,11 @@ func (r *MaasClusterReconciler) computeDesiredControlPlane(ctx context.Context,
 	return desiredCP, readyByKCP
 }
 
-// enoughCPNodesReady checks the target cluster for Ready control-plane nodes
-func (r *MaasClusterReconciler) enoughCPNodesReady(ctx context.Context, remoteClient client.Client, desiredCP, readyByKCP int32) bool {
+// enoughNodesReady checks the target cluster for Ready nodes (both control-plane and worker)
+func (r *MaasClusterReconciler) enoughNodesReady(ctx context.Context, remoteClient client.Client, desiredCP, readyByKCP int32) bool {
 	nodeList := &corev1.NodeList{}
-	cpSelector := labels.SelectorFromSet(labels.Set{
-		"node-role.kubernetes.io/control-plane": "",
-	})
-	if err := remoteClient.List(ctx, nodeList, &client.ListOptions{LabelSelector: cpSelector}); err == nil {
+	// Check all nodes, not just control-plane
+	if err := remoteClient.List(ctx, nodeList); err == nil {
 		ready := 0
```

Comment:
We are still using desiredCP count to check the stability of the cluster, but also including workers in the nodeList to be counted in ready. It will not give the desired output.

We can use totalDesiredNodes with totalReadyNodes OR keep it CP only calculation.